tests/lib/nested.sh

```diff
@@ -1230,6 +1229,44 @@ nested_exec_as() {
     sshpass -p "$PASSWD" ssh -p "$NESTED_SSH_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$USER"@localhost "$@"
 }
 
+nested_prepare_tools() {
+    if ! nested_exec "test -e retry" &>/dev/null; then
+        nested_copy "$TESTSTOOLS/retry"
+        nested_exec "sed 's/any-python/python3/g' -i ./retry"
+    fi
+
+    if ! nested_exec "test -e not" &>/dev/null; then
+        nested_copy "$TESTSTOOLS/not"
+    fi
+
+    if ! nested_exec "test -e MATCH" &>/dev/null; then
+        . "$TESTSLIB"/spread-funcs.sh
+        echo '#!/bin/bash' > MATCH_FILE
+        type MATCH | tail -n +2 >> MATCH_FILE
+        echo 'MATCH "$@"' >> MATCH_FILE
+        chmod +x MATCH_FILE
+        nested_copy "MATCH_FILE"
+        nested_exec "mv MATCH_FILE MATCH"
+        rm -f MATCH_FILE
+    fi
+
+    if ! nested_exec "test -e NOMATCH" &>/dev/null; then
+        . "$TESTSLIB"/spread-funcs.sh
+        echo '#!/bin/bash' > NOMATCH_FILE
+        type NOMATCH | tail -n +2 >> NOMATCH_FILE
+        echo 'NOMATCH "$@"' >> NOMATCH_FILE
+        chmod +x NOMATCH_FILE
+        nested_copy "NOMATCH_FILE"
+        nested_exec "mv NOMATCH_FILE NOMATCH"
+        rm -f NOMATCH_FILE
+    fi
+
+    if ! nested_exec "grep -qE PATH=.*/home/user1 /etc/environment"; then
+        # shellcheck disable=SC2016
+        nested_exec 'echo "PATH=/home/user1:$PATH" | sudo tee -a /etc/environment'
```

Comment:
Hm on a nested classic system we could put the tool under /usr/bin or somesuch. On a core we could try a location in /writable maybe? ($HOME is on writable, but maybe make that more official, eg. /writable/test-tools?)

Reply:
@bboozzoo sure, I will move it to /writable/test-tools
Thanks for the suggestion